src/Service/StorageAdapterPluginManagerFactory.php

```diff
@@ -9,6 +11,23 @@ final class StorageAdapterPluginManagerFactory
 {
     public function __invoke(ContainerInterface $container): AdapterPluginManager
     {
-        return new AdapterPluginManager($container);
+        $pluginManager = new AdapterPluginManager($container);
+
+        // If we do not have a config service, nothing more to do
+        if (! $container->has('config')) {
+            return $pluginManager;
+        }
+
+        $config = $container->get('config');
+
+        // If we do not have a configuration, nothing more to do
+        if (! isset($config[AdapterPluginManager::CONFIGURATION_KEY]) || ! is_array($config[AdapterPluginManager::CONFIGURATION_KEY])) {
+            return $pluginManager;
+        }
+
+        // Wire service configuration
+        $pluginManager->configure($config[AdapterPluginManager::CONFIGURATION_KEY]);
```

Comment:
You might want to store this value into a variable and use inline annotation to provide the `@var` type from https://github.com/laminas/laminas-servicemanager/blob/625f2aa3bc6dd02688b2da5155b3a69870812bda/src/ServiceManager.php#L84